modules/swagger-jaxrs/src/main/java/com/wordnik/swagger/jaxrs/Reader.java

```diff
@@ -70,9 +102,31 @@ public Reader(Swagger swagger, ReaderConfig config) {
     this.config = new DefaultReaderConfig(config);
   }
 
-  public Swagger read(Set<Class<?>> classes) {
+  public synchronized Swagger read(Set<Class<?>> classes) {
+
+    listeners.clear();
+
+    for(Class<?> cls: classes) {
+      if( ReaderListener.class.isAssignableFrom( ReaderListener.class ) && listeners.containsKey( cls )){
```

Comment:
This condition looks strange. The first part is always true, the second one is always false as the `listeners` map is empty after the `clean()` call.